src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Bases.cs

```diff
@@ -204,12 +202,22 @@ private SingleTypeDeclaration FirstDeclarationWithExplicitBases()
             return null;
         }
 
-        internal Tuple<NamedTypeSymbol, ImmutableArray<NamedTypeSymbol>> GetDeclaredBases(ConsList<Symbol> basesBeingResolved)
+        internal Tuple<NamedTypeSymbol, ImmutableArray<NamedTypeSymbol>> GetDeclaredBases(ConsList<Symbol> basesBeingResolved, bool ignoreNonNullTypesAttribute)
         {
             if (ReferenceEquals(_lazyDeclaredBases, null))
```

Comment:
Actually, we should continue to check `_lazyDeclaredBases` first, and only check `_lazyDeclaredBasesIgnoringNoNullTypesAttribute` if `_lazyDeclaredBases == null`.

---
In reply to: [201500168](https://github.com/dotnet/roslyn/pull/28320#discussion_r201500168) [](ancestors = 201500168)

Reply:
This doesn't appear to be resolved.

---
In reply to: [201714783](https://github.com/dotnet/roslyn/pull/28320#discussion_r201714783) [](ancestors = 201714783,201500168)

Reply:
Sorry, I'm confused. What you describe seems like what I'm already doing: only if `_lazyDeclaredBases` is `null` and `ignoreNonNullTypesAttribute` is set do we bother with computing or reading `_lazyDeclaredBasesIgnoringNoNullTypesAttribute`.

Reply:
Shouldn't we also check `ReferenceEquals(_lazyDeclaredBasesIgnoringNonNullTypesAttribute, null)`, to avoid calling `MakeDeclaredBases()` unnecessarily?

---
In reply to: [202064006](https://github.com/dotnet/roslyn/pull/28320#discussion_r202064006) [](ancestors = 202064006)